torchao/quantization/quantize_/workflows/float8/float8_tensor.py

```diff
@@ -0,0 +1,609 @@
+    if block_size_for_dim == 1:
+        # Scale is per-element along this dimension
+        # Slice away as normal
+        return aten.slice.Tensor(scale, dim, start, end, step)
+    else:
+        # There is blocking in this dimension
+        # Calculate which scale elements correspond to the sliced data
+        scale_start = start // block_size_for_dim if start is not None else None
+        scale_end = (
+            (end + block_size_for_dim - 1) // block_size_for_dim
+            if end is not None
+            else None
+        )
+
+        # Error on Step > 1
+        if step > 1:
+            raise NotImplementedError(
+                "Slicing with step > 1 is not implemented for scale tensors."
+            )
+
+        return aten.slice.Tensor(scale, dim, scale_start, scale_end, 1)
+
+
+@dataclass
+class QuantizeTensorToFloat8Kwargs(QuantizeTensorKwargs):
+    """Tensor kwargs for creating float8 tensor (either activation or weight)
+
+    Args:
+       dtype (torch.dtype): the dtype for float8 Tensor
+       granularity (FP8Granularity): the granularity for the Tensor, currently either PerRow() or PerTensor()
+    """
+
+    dtype: torch.dtype = torch.float8_e4m3fn
+    granularity: FP8Granularity = PerRow()
+    mm_config: Float8MMConfig = Float8MMConfig(use_fast_accum=True)
+
+
+class Float8Tensor(TorchAOBaseTensor):
+    """
+    Float8 Quantized (weight) Tensor, with float8 dynamic quantization for activation or bfloat16 activation.
+
+    TODO: needs padding for cutlass kernels
+
+    Tensor Attributes:
+        _data: float8 raw data
+        scale: the scale for float8 Tensor
+
+    Non-Tensor Attributes:
+        block_size (List[int]): the block size for float8 quantization, meaning the shape of the elements
+        sharing the same set of quantization parameters (scale), have the same rank as _data or
+        is an empty list (representing per tensor quantization)
+        act_quant_kwargs (QuantizeTensorToFloat8Kwargs): the kwargs for Float8Tensor.to_float8
+        dtype: Original Tensor dtype
+    """
+
+    tensor_data_attrs = ["_data", "scale"]
+    tensor_attributes = ["block_size", "mm_config", "act_quant_kwargs", "dtype"]
+
+    def __new__(
+        cls,
+        _data,
+        scale,
+        block_size,
+        mm_config,
+        act_quant_kwargs,
+        dtype,
+    ):
+        shape = _data.shape
+        kwargs = {}
+        kwargs["device"] = _data.device
+        kwargs["dtype"] = dtype
+        kwargs["requires_grad"] = False
+        return torch.Tensor._make_wrapper_subclass(cls, shape, **kwargs)  # type: ignore[attr-defined]
+
+    def __init__(
+        self,
+        _data: torch.Tensor,
+        scale: torch.Tensor,
+        block_size: Optional[List[int]] = None,
+        mm_config: Float8MMConfig = Float8MMConfig(use_fast_accum=True),
+        act_quant_kwargs: Optional[QuantizeTensorToFloat8Kwargs] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        self._data = _data
```

Comment:
nit: consistency with leading underscore, either all attributes have it or all don't?

Reply:
I actually prefer to not have underscore everywhere, does `qdata` look good? if not, I can use underscore everywhere